compiler/rustc_hir_typeck/src/fn_ctxt/adjust_fulfillment_errors.rs

```diff
@@ -95,6 +128,51 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
 
         match self.tcx.hir_node(hir_id) {
+            hir::Node::Expr(&hir::Expr {
```

Comment:
vibe: please move all `hir::Node::Expr` branches into a submatch in a sub-fn, and start that fn by checking `self.closure_span_overlaps_error(error, span)`. This function is currently very difficult to parse for me